### ISSUE TITLE
Change to branching workflow

### ISSUE THREAD

**Issue by TheRealRavager** [AY1920S1-CS2103T-T13-1/main#39]
Changing to branching workflow to lessen administrative workload.

Please indicate once all your work has been migrated into Team repo's branches

- [x] @C-likethis123 
- [x] @TheRealRavager 
- [x] @WilliamRyank 
- [x] @SimNee 
- [x] @Yen-Peng 

DO NOT DELETE YOUR FORK AS IT IS USED FOR GRADING

### How to migrate your work:

1. Clone the **TEAM** repo.
2. Create a new branch with the same name as the branch in your fork. e.g. if your fork's branch is called `refactor-person-class`, create a new branch on the team repo with the exact same name.
3. Create a PR to merge your fork's branch into the **branch** on the **Team** repo.
Example shown below:
![image](https://user-images.githubusercontent.com/43085796/66105013-dc258600-e5ec-11e9-90d9-bd3a24fe4626.png)
4. Once merged, work ONLY on your branch on the team repo

### How to do branching workflow:
1. Change to master branch on *TEAM* repo.
2. Verify you are on master by doing `git status`
3. Get the latest updates on master by doing `git pull`
4. Branch off master by doing `git checkout -b your-branch-name`
5. Commit as per usual until you are done, then make a PR into master from your branch.
Example shown below:

*Change both base repositories to the team's repo*
![image](https://user-images.githubusercontent.com/43085796/66107833-c8315280-e5f3-11e9-9a71-2f974d8a337d.png)

*It will automatically "collapse" into the following form:*
![image](https://user-images.githubusercontent.com/43085796/66107887-e6974e00-e5f3-11e9-9285-9ef9befe1e27.png)

*Change to the correct branch*
![image](https://user-images.githubusercontent.com/43085796/66107924-f747c400-e5f3-11e9-88ad-21597e480a06.png)

You are done! Congrats you have now switched to the branching workflow!

## Branch Protection
Currently, only Master requires approvals for PR before merge. However, please refrain from pushing to other people's branch (its quite rude). If you want to help, branch off their branch and make a PR. Merge only when branch owner has given their approval.  

TL;DR
- Dont push to other people's branch
- If you want to help:
    - Branch off their branch
    - PR into their branch
    - Merge only when they say its ok

**Comment by TheRealRavager** [comment]
Feel free to ask if any qns, either here or on Team Telegram

**Comment by TheRealRavager** [comment]
@SimNee @WilliamRyank @C-likethis123 @Yen-Peng can yall comment here once you read this guide. If youre the last person to comment, please also close the issue to reduce clutter. Thanks!
